src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -475,247 +401,37 @@ def _create_meta_lookup(root_dir, env, kbase_collection, load_ver, tool):
-            categories[trait_category] = ColumnCategory(category=trait_category, columns=[])
-
-        # add the trait to the appropriate category's list of columns
-        categories[trait_category].columns.append(ColumnInformation(
-            col_id=str(trait[_SYS_TRAIT_INDEX]),
-            name=trait[_SYS_TRAIT_NAME],
-            description=trait[_SYS_TRAIT_DESCRIPTION],
-            type=trait_type
-        ))
-
-    # sort columns in each ColumnCategory object by the column id
-    for category in categories.values():
-        category.columns = sorted(category.columns, key=lambda column: int(column.col_id))
-
-    # sort ColumnCategory objects by the column id of the first column in each ColumnCategory object
-    sorted_categories = sorted(
-        categories.values(), key=lambda category: int(category.columns[0].col_id)
-    )
-
-    # this is to ensure that the column ids are in ascending order
-    # TODO: might need to skip this step for heatmap products other than microtrait
-    column_ids = [column.col_id for category in sorted_categories for column in category.columns]
-    if not _ensure_list_ordered(column_ids):
-        raise ValueError(f'Column ids are not ordered in ascending order: {column_ids}')
-
-    return sorted_categories
-
-
-def _is_float_int(num: Any) -> bool:
-    # Check if a number is an integer or a float with an integer value
-    return isinstance(num, int) or (isinstance(num, float) and num.is_integer())
-
-
-def _int(num: float | int) -> int:
-    # Convert a float to an integer if the float is an integer
-    if _is_float_int(num):
-        return int(num)
-    else:
-        raise ValueError(f'Input must be an integer. Got {num} instead.')
-
+def _build_heatmap_meta(
+        reference_meta: list[dict],
+        kbase_collection: str,
+        load_ver: str,
+        min_value: float,
+        max_value: float) -> dict:
+    # Build the heatmap metadata from the reference metadata (list of metadata)
 
-def _is_binary(num: int) -> bool:
-    # Given a number, checks if it is a binary num (i.e., being only 0 or 1).
-    return num == 0 or num == 1
+    heatmap_categories = dict()
+    for meta in reference_meta:
+        category = meta[FIELD_HEATMAP_CATEGORY]
+        if category not in heatmap_categories:
+            heatmap_categories[category] = {FIELD_HEATMAP_CATEGORY: category, FIELD_HEATMAP_COLUMNS: []}
 
+        meta_without_category = meta.copy()
+        del meta_without_category[FIELD_HEATMAP_CATEGORY]
+        heatmap_categories[category][FIELD_HEATMAP_COLUMNS].append(meta_without_category)
 
-def _num_to_bool(num: int | bool) -> bool:
-    # Given a number, checks if it is a binary type (i.e., contains only 0s and 1s).
+    # sort the columns by column id
+    for category in heatmap_categories.values():
+        category[FIELD_HEATMAP_COLUMNS] = sorted(category[FIELD_HEATMAP_COLUMNS],
+                                                 key=lambda x: int(x[FIELD_HEATMAP_COL_ID]))
```

Comment:
We do an equivalent sort for the rows in the tool container right?

Reply:
no. But It's good idea to sort it there. I added the sort in tool runner.

Reply:
:+1:

Reply:
It looks like you took out the sort in the parser, are you just relying on the tool runner sort to make sure the meta columns are ordered?

Reply:
yea. I think it makes sense to sort the pre-processed results in tool runner. Do you think it's necessary to double check in the global parser?

Reply:
Nope, that seems fine, I just wanted to make sure I understood the change

:+1: